lib/Dialect/ModuleAttributes.cpp

```diff
@@ -31,6 +34,27 @@ bool moduleIsCGGI(Operation *moduleOp) {
          nullptr;
 }
 
+Attribute getSchemeParamAttr(Operation *op) {
+  SmallVector<StringLiteral> schemeAttrNames = {
+      bgv::BGVDialect::kSchemeParamAttrName,
+      ckks::CKKSDialect::kSchemeParamAttrName,
+  };
+
+  Operation *moduleOp = op;
+  if (!isa<ModuleOp>(op)) {
+    moduleOp = op->getParentOfType<ModuleOp>();
+  }
+
+  for (auto schemeAttrName : schemeAttrNames) {
+    if (auto schemeAttr = moduleOp->getAttr(schemeAttrName)) {
+      return schemeAttr;
+    }
+  }
+
+  // A default case to avoid nullptr errors
+  return UnitAttr::get(op->getContext());
```

Comment:
I actually always use `nullptr` for empty Attribute and assume attribute could be nullptr everywhere and add checks there. I think this is rather standard? https://github.com/llvm/llvm-project/blob/65cede26a6b06ba02c08284fada06c46c0289704/mlir/include/mlir/IR/Attributes.h#L38-L48

Reply:
I had considered this, and then I was uncertain what would happen if you pass the result to a type switch. I will give it a try though.

Reply:
Using nullptr causes TypeSwitch to stack trace when the module does not have an attribute attached. This is because TypeSwitch internally uses dyn_cast instead of dyn_cast_or_null.

Reply:
I ended up continuing to use UnitAttr, but having the TypeSwitches that use it return a `FailureOr<T>` for more graceful failures.